python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -122,62 +122,53 @@ async def get_chat_message_contents(
         Returns:
             A list of chat message contents.
         """
+        if (
+            settings.function_choice_behavior is None
+            or not settings.function_choice_behavior.auto_invoke_kernel_functions
+        ):
+            return await self._send_chat_request(chat_history, settings)
+
+        self._verify_function_choice_behavior(settings, **kwargs)
```

Comment:
How about if we remove the need to get the kernel/arguments from kwargs twice? Something like:

```python
self._verify_function_choice_behavior(settings, kernel=kwargs.get("kernel"), arguments=kwargs.get("arguments"))
```
and then:
```python
def _verify_function_choice_behavior(
    self, 
    settings: AzureAIInferenceChatPromptExecutionSettings, 
    kernel: Kernel, 
    arguments: KernelArguments
):
    """Verify the function choice behavior."""
    if settings.function_choice_behavior is not None:
        if kernel is None:
            raise ServiceInvalidExecutionSettingsError("Kernel is required for tool calls.")
        if arguments is None and settings.function_choice_behavior.auto_invoke_kernel_functions:
            raise ServiceInvalidExecutionSettingsError("Kernel arguments are required for auto tool calls.")
        if settings.extra_parameters is not None and settings.extra_parameters.get("n", 1) > 1:
            # Currently only OpenAI models allow multiple completions but the Azure AI Inference service
            # does not expose the functionality directly. If users want to have more than 1 response, they
            # need to configure `extra_parameters` with a key of "n" and a value greater than 1.
            raise ServiceInvalidExecutionSettingsError(
                "Auto invocation of tool calls may only be used with a single completion."
            )
```